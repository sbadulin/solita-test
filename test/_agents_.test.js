import superagent from "superagent";

const API_ROOT = "http://localhost:8484/";

const responseBody = res => res.body || res.text;

const requests = {
  get: (url, query) =>
    superagent
      .get(`${API_ROOT}api${url}`)
      .query(query)
      .then(responseBody)
};

const productsAgent = {
  get: productId =>
    productId
      ? requests.get(`/products/${productId}`)
      : requests.get(`/products`)
};

describe("products agent", () => {
  it("get all products from API", async () => {
    expect.assertions(1);
    try {
      const { products } = await productsAgent.get();
      expect(products).toMatchSnapshot();
    } catch (error) {
      throw Error(error);
    }
  }, 20000);
});
