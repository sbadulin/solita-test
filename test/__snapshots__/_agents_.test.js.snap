// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products agent get all products from API 1`] = `
Array [
  Object {
    "description": "Perfect choice when you don’t need any corners.",
    "id": 1,
    "name": "Circle",
    "price": 999,
  },
  Object {
    "description": "For once, it’s a great idea to think inside the box.",
    "id": 2,
    "name": "Rectangle",
    "price": 899,
  },
  Object {
    "description": "A true classic with three elegant corners.",
    "id": 3,
    "name": "Triangle",
    "price": 1009,
  },
]
`;
